server/etcdserver/api/v3rpc/auth.go

```diff
@@ -164,3 +165,21 @@ func (as *AuthServer) UserChangePassword(ctx context.Context, r *pb.AuthUserChan
 	}
 	return resp, nil
 }
+
+type AuthGetter interface {
+	AuthInfoFromCtx(ctx context.Context) (*auth.AuthInfo, error)
+	AuthStore() auth.AuthStore
+}
+
+type AuthChecker struct {
```

Comment:
```suggestion
type RequireAdmin struct {
```
This auth checker ony checks admin permission.